crates/sqlexec/src/session.rs

```diff
@@ -740,25 +734,46 @@ impl Session {
         }
     }
 
-    pub fn parse_query(&mut self, query: &str) -> Result<VecDeque<StatementWithExtensions>> {
+    /// Create a logical plan from a SQL query.
+    /// if the query doesn't contain exactly one statement, an error is returned.
+    pub async fn create_logical_plan(&mut self, query: &str) -> Result<LogicalPlan> {
```

Comment:
this was the cause of the metastore issue. I was creating the plan without first updating the state, so the tables "werent there" because the catalog state was outdated.